CopyKeys/CopyKeys.ps1

```diff
@@ -1,1468 +1,1676 @@
+                $LocationDropDown.Font).Width, $LocationDropDown.Width | Measure-Object -Maximum).Maximum
+        }
+
+        for ($Index = 4; $Index -lt $FormElementsList.Count; $Index++)
+        {
+            $FormElements[$FormElementsList[$Index]].Enabled = $false
+        }
+
+        $LoadingLabel.Text = [string]::Empty
+    }
+}
+
+### <summary>
+### Gets list of VMs for selected resource group and populates checklist.
+### </summary>
+function Get-VirtualMachines
+{
+    $ResourceGroupName = $this.SelectedItem.ToString()
+    if ($ResourceGroupName)
+    {
+        $LoadingLabel.Text = [string]::Empty
+        $VmListBox = $FormElements["VmListBox"]
+        $VmListBox.Items.Clear()
+        $FormElements["VmLabel"].Enabled = $true
+        $FormElements["LocationLabel"].Enabled = $true
+        $VmListBox.Enabled = $true
+        $LocationDropDown.Enabled = $true
+        $LocationDropDown.Text = [string]::Empty
+
+        $VmList = (Get-AzVm -ResourceGroupName $ResourceGroupName) | Sort-Object Name
+
+        foreach ($Item in $VmList)
+        {
+            if (($null -ne $Item.Extensions) -and
+                ($Item.Extensions.Id | ForEach-Object { `
+                    $_.split('/')[-1].tolower().contains( `
+                        [ConstantStrings]::adeExtensionPrefix)}) -contains $true)
+            {
+                $SuppressOutput = $VmListBox.Items.Add($Item.Name)
+            }
+        }
+
+        if($VmList -and ($VmListBox.Items.Count -gt 0))
+        {
+            $Longest = ($VmList.Name | Sort-Object Length -Descending)[0]
+            $Size = [System.Windows.Forms.TextRenderer]::MeasureText($Longest, `
+                $VmListBox.Font).Width
+
+            if ($Size -gt $VmListBox.Width)
+            {
+                $VmListBox.Width = $Size + 30
+                $UserInputForm.Width = $Size + 60
+            }
+        }
+        else
+        {
+            $LoadingLabel.Text = [ConstantStrings]::noAdeVmInResourceGroup
+        }
+
+        for ($Index = 8; $Index -lt $FormElementsList.Count; $Index++)
+        {
+            $FormElements[$FormElementsList[$Index]].Enabled = $false
+        }
+    }
+}
+
+### <summary>
+### Disable and clears remaining options when VM list modified.
+### </summary>
+function Disable-RestOfOptions
+{
+    $FormElements["LocationDropDown"].Text = [string]::Empty
+    $FormElements["BekDropDown"].Text = [string]::Empty
+    $FormElements["KekDropDown"].Text = [string]::Empty
+
+    for ($Index = 8; $Index -lt $FormElementsList.Count; $Index++)
+    {
+        if ($FormElements[$FormElementsList[$Index]].Text -ne [ConstantStrings]::notApplicable)
+        {
+            $FormElements[$FormElementsList[$Index]].Enabled = $false
+        }
+    }
+}
+
+### <summary>
+### Gets list of target key vaults for KEK and BEK for selected VM(s) and populates dropdown.
+### </summary>
+function Get-KeyVaults
```

Comment:
One request which came from customers is :
Key vaults need not be same region as vm, so can i choose a custom target key vault in another region

Reply:
Previously we were giving that option but now are restricting it.